### PR TITLE
Adding support for environment argument

### PR DESCRIPTION
**Pull request by bbergstrom**
* Add environment argument to deploy command for convenience and in preparation for non-interactive support.
* Fixed a bunch of test issues and expanded some coverage.
* Added exception message to output on invalid config so the user gets some idea of why its invalid.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -21,3 +21,10 @@
   #     --seed 1234
   #config.order = 'random'
 end
+
+CONFIG_PATHS = {
+    valid_modern: File.expand_path("../files/soyuz_valid_modern.yml", __FILE__),
+    valid_legacy: File.expand_path("../files/soyuz_valid_legacy.yml", __FILE__),
+    incompatible: File.expand_path("../files/soyuz_incompatible.yml", __FILE__),
+    invalid: File.expand_path("../files/soyuz_invalid.yml", __FILE__),
+}
```
Comment (NickLaMuro) [inline spec/spec_helper.rb]: Four space tabbing... really!?  What is this, C?

Comment (bbergstrom) [inline spec/spec_helper.rb]: 2 space indent. 4 space continuation indent.

File: lib/soyuz/config.rb [hunk 2]
```diff
@@ -35,16 +35,16 @@ def config_data
     end
 
     def load_config
+      File.open(@config_file, 'r:bom|utf-8')
```
Comment (NickLaMuro) [inline lib/soyuz/config.rb]: I assume the point of this is to error out quicker if it isn't the correct file type.

Comment (bbergstrom) [inline lib/soyuz/config.rb]: OH NOES.  This was part of my debugging code.  I will remove this line.

**Pull request merged**